node/gum/src/lib.rs

```diff
@@ -112,7 +112,53 @@ pub use jaeger::hash_to_trace_identifier;
 #[doc(hidden)]
 pub use polkadot_primitives::{CandidateHash, Hash};
 
-pub use gum_proc_macro::{debug, error, info, trace, warn};
+pub use gum_proc_macro::{debug, error, info, trace, warn, warn_if_frequent};
 
 #[cfg(test)]
 mod tests;
+
+/// Max allowed frequency of debug messages
+pub const MAX_FREQ_RATE: f64 = 1.0;
+const MAX_FREQ_SIZE: usize = 10;
+
+/// Utility struct to compare the rate of its own calls.
+pub struct Freq {
+	timestamps: Vec<u64>,
+}
+
+impl Freq {
+	/// Initiates a new instanse
+	pub fn new() -> Self {
+		Self { timestamps: Vec::with_capacity(MAX_FREQ_SIZE) }
+	}
+
+	/// Compares the rate of its own calls with the passed one.
+	pub fn is_frequent(&mut self, max_rate: f64) -> bool {
+		self.record();
+
+		// Two attempts is not enough to call something as frequent.
+		if self.timestamps.len() < 3 {
+			return false
+		}
+
+		let elapsed = self.timestamps.last().unwrap() - self.timestamps.first().unwrap();
+		if elapsed == 0 {
```

Comment:
Would be slightly better to use `checked_div` below and then match on the `Option`.

Reply:
Sorry, what do you mean?

Reply:
ah, I see. But we don't need it anymore since we use floats division.